### DISCUSSION TITLE
Should we block new ScaledObjects for scale targets who already have a linked ScaledObject?

### DISCUSSION THREAD

**Discussion by tomkerkhove** [kedacore/keda#3023] [Ideas]
As we recommend to only using one ScaledObject per scale target, I was wondering if we should introduce validation (if not already) to avoid having multiple autoscaling definitions for the same workload?

If we do this, do we do it as part of KEDA core or make this something cluster operators can configure (~ autoscaling governance/control)?

**Comment by zroubalik** [comment]
This would be nice addition, but not something that should be started by default imho.

Do you see it just a check that there aren't multiple ScaledObjects assigned to the workload? Or do you want to also include arbitrary HPAs?

**Reply by tomkerkhove** [reply]
I think both are valid checks, but multiple SO's would be an easy start.

For multiple SO's I would block the creating of the second SO. For arbitrary HPA I would log warnings given we don't control that; but I might be too strict here.

> This would be nice addition, but not something that should be started by default IMHO.

Agreed! At least not for 2.X.

**Reply by zroubalik** [reply]
Agree, let's convert this into a proper issue.

**Reply by tomkerkhove** [reply]
I'll open one later today.

**Reply by tomkerkhove** [reply]
Created https://github.com/kedacore/keda/issues/3083

> Lastly, if somebody creates an HPA while it is already being autoscaled by an ScaledJob/ScaledObject we should log a warning in KEDA so that they become aware of it. This is because we cannot reject HPA resources

I might be going too far, but what if we introduce an admission controller to reject the HPA in this case it would be nice IMO.

Thoughts?

**Reply by zroubalik** [reply]
Yeah, that might be a way :)

**Reply by tomkerkhove** [reply]
https://github.com/kedacore/keda/issues/3087
